controllers/disruption_cron_controller.go

```diff
@@ -57,11 +62,42 @@ func (r *DisruptionCronReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	r.log.Infow("fetched last known history", "history", instance.Status.History)
 	DisruptionCronTags = []string{"disruptionCronName:" + instance.Name, "disruptionCronNamespace:" + instance.Namespace, "targetName:" + instance.Spec.TargetResource.Name}
 
+	// check whether the object is being deleted or not
 	if !instance.DeletionTimestamp.IsZero() {
-		// Add finalizer here if required
+		// the instance is being deleted, clean it if the finalizer is still present
+		if controllerutil.ContainsFinalizer(instance, chaostypes.DisruptionCronFinalizer) {
+			controllerutil.RemoveFinalizer(instance, chaostypes.DisruptionCronFinalizer)
+
+			// we reach this code when all the cleanup disruption cron have succeeded
+			// we can remove the finalizer and let the resource being garbage collected
+			if err := r.Client.Update(ctx, instance); err != nil {
+				return ctrl.Result{}, fmt.Errorf("error removing disruptioncron finalizer: %w", err)
+			}
+
+			if err := r.emmitEvent(instance, chaosv1beta1.EventDisruptionCronDeleted); err != nil {
+				r.log.Warnw("failed to emit event", "err", err)
+			}
+		}
+
 		return ctrl.Result{}, nil
 	}
 
+	if controllerutil.ContainsFinalizer(instance, chaostypes.DisruptionCronFinalizer) {
+		if err := r.emmitEvent(instance, chaosv1beta1.EventDisruptionCronUpdated); err != nil {
```

Comment:
We're going to send this event every time the Disruption is reconciled without a deletion timestamp, except for the very first reconciliation. That removes any use of this to indicate that a user updated the resource. If we wanted to count reconciliations, a metric would be better.

Reply:
By adding the hash spec of the disruption in the message, the event will be not dropped only if the spec has changed, and we want to track those kind of changes.